frontend/src/modules/fluenteditor/components/RichTranslationForm.js

```diff
@@ -327,26 +328,38 @@ export class RichTranslationFormBase extends React.Component<InternalProps> {
         />;
     }
 
-    renderElements(
-        elements: Array<FluentElement>,
-        path: MessagePath,
-        label: string,
-    ): React.Node {
+    renderItem(value: string, path: MessagePath, label: string) {
+        return <tr key={ `${path.join('-')}` }>
+            <td>
+                <label htmlFor={ `${path.join('-')}` }>{ label }</label>
+            </td>
+            <td>
+                { this.renderInput(value, path) }
+            </td>
+        </tr>;
+    }
+
+    renderElements(elements: Array<FluentElement>, path: MessagePath, label: string): React.Node {
         return elements.map((element, index) => {
-            if (element.type !== 'TextElement') {
-                return null;
+            if (element.type === 'Placeable' && element.expression.type === 'SelectExpression') {
```

Comment:
Here and below, there's a somewhat mixed up assumption that `flattenMessage` has been called. On one hand, you're testing the inside of the `Placeable`, and on the other hand, you're assuming it's a `TextElement` if it's not a `SelectExpression`.

That puzzles me a bit, but I don't have a good proposal on how to resolve that.

Reply:
Here (and in `RichString.js ` as well) we expect `flattenMessage` to already been called. That's why we can hardcode `0` in `variant.value.elements[0].value`. For `SelectExpressions` we only flatten variants.